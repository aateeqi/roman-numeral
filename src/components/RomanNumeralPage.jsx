import { extractParamFromQueryString } from "../utils/extractParamFromQueryString";
import { getRomanNumeral } from "../service/romanNumeralService";

export const RomanNumeral = (props) => {
  let query = "";
  if (props && props.location) {
    query = props.location.search;
  }
  const number = extractParamFromQueryString(query, "query");
  const romanNumeral = getRomanNumeral(number);
  if (!number || !romanNumeral) {
    return <h1>Your parameters are invalid</h1>;
  }

  return (
    <h1>
      Roman Numeral for {number} is {romanNumeral}
    </h1>
  );
};
