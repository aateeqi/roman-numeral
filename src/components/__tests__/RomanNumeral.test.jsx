import React from "react";
import renderer from "react-test-renderer";
import { RomanNumeral } from "../RomanNumeralPage";

test("renders valid input correctly", () => {
  const romanNumeralMockProps = {
    location: {
      search: "?query=999",
    },
  };
  const tree = renderer
    .create(<RomanNumeral {...romanNumeralMockProps} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders invalid input correctly", () => {
  const romanNumeralMockProps = {
    location: {
      search: "?query=-190",
    },
  };
  const tree = renderer
    .create(<RomanNumeral {...romanNumeralMockProps} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders when no location exists", () => {
  const tree = renderer.create(<RomanNumeral />).toJSON();
  expect(tree).toMatchSnapshot();
});
