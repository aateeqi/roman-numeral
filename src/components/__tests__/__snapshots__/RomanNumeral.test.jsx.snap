// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders invalid input correctly 1`] = `
<h1>
  Your parameters are invalid
</h1>
`;

exports[`renders valid input correctly 1`] = `
<h1>
  Roman Numeral for 
  999
   is 
  CMXCIX
</h1>
`;

exports[`renders when no location exists 1`] = `
<h1>
  Your parameters are invalid
</h1>
`;
